content/backend/typescript-apollo/0-introduction.md

```diff
@@ -0,0 +1,130 @@
+  ]
+correctAnswer: 3
+---
+
+## Overview
+
+GraphQL is a rising star in the domain of API development. It is gaining significant ground over REST as an API design paradigm and is becoming one of the
+standards for exposing the data and functionality of a web server.
+
+In this tutorial, you'll learn how to build an _idiomatic_ GraphQL server entirely from scratch. You are going to use
+the following technologies:
+
+- [Typescript](https://www.typescriptlang.org/) - Strongly typed superset of Javascript that can be transpiled back to Javascript.  Typescript has enjoyed significant adoption and love in the developer community for the type-safety and improved developer experience it provides. 
+- [Apollo Server](https://github.com/apollographql/apollo-server/tree/main/packages/apollo-server): Fully-featured GraphQL Server with focus on easy setup, performance and great developer experience,
+- [Nexus](https://github.com/graphql-nexus/nexus): A library for creating robust, type-safe GraphQL Schema using Javascript/Typescript. 
+- [Prisma](https://www.prisma.io/): Next-generation Node.js and Typescript ORM. You can use the Prisma Client to access your database inside of
+  GraphQL resolvers.
+
+
+
+## What this tutorial will cover
+
+The goal of this tutorial is to build an API for a [Hacker News](https://news.ycombinator.com/) clone. Here is a quick
+rundown of what to expect.
+
+You'll start by learning the basics of how a GraphQL server works, start defining a
+[_GraphQL schema_](https://www.prisma.io/blog/graphql-server-basics-the-schema-ac5e2950214e) for your server using [Nexus](https://github.com/graphql-nexus/nexus) and writing
+corresponding _resolver functions_. In the beginning, these resolvers will only work with data that's stored in-memory -
+so nothing will persist beyond the runtime of the server.
+
+Nobody wants a server that's not able to store and persist data, right? Not to worry! Afterwards, you're going to add a
+[SQLite](http://sqlite.org/) database to the project which will be managed with [Prisma](https://www.prisma.io/).
+
+Once you have the database connected, you are going to add more advanced features to the API.
+
+You'll start by implementing signup/login functionality that enables users to authenticate against the API. This will also allow you to check the permissions of your users for certain API operations.
+
+Then, you're going to add a vote feature. 
+
+Finally, you'll allow the consumers of the API to constrain the list of items they retrieve from the API by adding sorting, filtering and pagination capabalities to it.
+
+Throughout the tutorial you will learn GraphQL bit by bit, through _theory, writing code and reading external resources_. You'll learn not just the _how_ of various technology choices, but also the _why_, including their pros, cons and tradeoffs. 
+
+
+
+## Conventions and Assumptions
+
+There's a few conventions and assumptions followed throughout this tutorial that you should know about.
```

Comment:
I would mention that reading the following sections is somewhat optional and if people are keen to get started they can immediately skip to the next page.